<p>Hello John, I wanted to thank you for the work you’ve done teaching Angular. I’ve been using your Pluralsight courses and your github repos as a resource as I continue to learn the framework.</p>

<p>I wanted to ask you a question about something I found in https://github.com/johnpapa/ng-demos/blob/master/modular/src/client/app/core/dataservice.js</p>

<p>It seems to me, when you return the promise from the catch() in both getAvengers() and getAvengersCount(), that promise will <em>always</em> be resolved (and not rejected). This is because each then or catch in a promise chain returns a new promise that is only rejected if you throw or return a rejected promise from one of your handlers. If you attempt to chain off that promise, as you do in https://github.com/johnpapa/ng-demos/blob/master/modular/src/client/app/avengers/avengers.js, the success callback is always called. In this particular case, the data is undefined, and nothing depends on there being data so it works.</p>

<p>Was this your intention?
<!--more-->
AUTHOR: Bert Evans
AUTHOR EMAIL: alevans4@gmail.com
AUTHOR URL: 
SUBJECT: [John Papa] Connect
IP: 69.245.11.195
Array
(
    [1_Name] =&gt; Bert Evans
    [2_Email] =&gt; alevans4@gmail.com
    [3_Website] =&gt; 
    [4_Comment] =&gt; Hello John, I wanted to thank you for the work you’ve done teaching Angular. I’ve been using your Pluralsight courses and your github repos as a resource as I continue to learn the framework.</p>

<p>I wanted to ask you a question about something I found in https://github.com/johnpapa/ng-demos/blob/master/modular/src/client/app/core/dataservice.js</p>

<p>It seems to me, when you return the promise from the catch() in both getAvengers() and getAvengersCount(), that promise will <em>always</em> be resolved (and not rejected). This is because each then or catch in a promise chain returns a new promise that is only rejected if you throw or return a rejected promise from one of your handlers. If you attempt to chain off that promise, as you do in https://github.com/johnpapa/ng-demos/blob/master/modular/src/client/app/avengers/avengers.js, the success callback is always called. In this particular case, the data is undefined, and nothing depends on there being data so it works.</p>

<p>Was this your intention?
)</p>

